### ISSUE TITLE
dry-run on kubernetes provider does not work

### ISSUE THREAD

**Issue by charliedrage** [projectatomic/atomicapp#339]
```
dropbox/dev/atomicapp  remove-test-file ✗                                                                              14m ◒  
▶ atomicapp -v --dry-run stop tests/units/cli/test_examples/helloapache
2015-10-19 10:44:47,624 - atomicapp.plugin - DEBUG - Loading providers from /usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/providers
2015-10-19 10:44:47,634 - atomicapp.plugin - DEBUG - Found provider <class 'kubernetes.KubernetesProvider'>
2015-10-19 10:44:47,634 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
2015-10-19 10:44:47,634 - atomicapp.plugin - DEBUG - Found provider <class 'kubernetes.KubernetesProvider'>
2015-10-19 10:44:47,635 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
2015-10-19 10:44:47,635 - kubernetes - DEBUG - Given config: {u'image': u'centos/httpd', u'hostport': 80, 'namespace': 'default', 'provider': 'kubernetes'}
2015-10-19 10:44:47,635 - kubernetes - INFO - Using namespace default
2015-10-19 10:44:47,635 - kubernetes - INFO - trying kubectl at /usr/bin/kubectl
2015-10-19 10:44:47,635 - kubernetes - INFO - trying kubectl at /usr/local/bin/kubectl
2015-10-19 10:44:47,635 - kubernetes - INFO - found kubectl at /usr/local/bin/kubectl
2015-10-19 10:44:47,635 - kubernetes - INFO - Undeploying from Kubernetes
2015-10-19 10:44:47,635 - kubernetes - DEBUG - tests/units/cli/test_examples/helloapache/artifacts/kubernetes/.hello-apache-pod.json
2015-10-19 10:44:47,651 - kubernetes - DEBUG - stdout = No resources found

2015-10-19 10:44:47,652 - kubernetes - DEBUG - stderr = error: could not read an encoded object from tests/units/cli/test_examples/helloapache/artifacts/kubernetes/.hello-apache-pod.json: unable to connect to a server to handle "pods": couldn't read version from server: Get http://localhost:8080/api: dial tcp 127.0.0.1:8080: connection refused


2015-10-19 10:44:47,652 - atomicapp.utils - INFO - atomicapp.status.error.message=cmd failed: /usr/local/bin/kubectl --kubeconfig=provider.config delete -f tests/units/cli/test_examples/helloapache/artifacts/kubernetes/.hello-apache-pod.json --namespace=default
2015-10-19 10:44:47,652 - atomicapp.cli.main - ERROR - error: could not read an encoded object from tests/units/cli/test_examples/helloapache/artifacts/kubernetes/.hello-apache-pod.json: unable to connect to a server to handle "pods": couldn't read version from server: Get http://localhost:8080/api: dial tcp 127.0.0.1:8080: connection refused

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/cli/main.py", line 67, in cli_stop
    NuleculeManager.do_stop(**vars(args))
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/nulecule/main.py", line 44, in do_stop
    m.stop(APP, cli_provider, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/nulecule/main.py", line 128, in stop
    self.nulecule.stop(cli_provider, dryrun)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/nulecule/base.py", line 114, in stop
    component.stop(provider_key, dryrun)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/nulecule/base.py", line 212, in stop
    provider.undeploy()
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/providers/kubernetes.py", line 224, in undeploy
    self._call(cmd)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/providers/kubernetes.py", line 129, in _call
    raise Exception(str(stderr))
Exception: error: could not read an encoded object from tests/units/cli/test_examples/helloapache/artifacts/kubernetes/.hello-apache-pod.json: unable to connect to a server to handle "pods": couldn't read version from server: Get http://localhost:8080/api: dial tcp 127.0.0.1:8080: connection refused
```

__UPDATE__:

Actually, --dry-run on *anything* with `stop` doesn't work.
try `atomicapp -v --dry-run stop --provider=docker tests/units/cli/test_examples/helloapache`

**Comment by charliedrage** [comment]
Okay, looks like I found what the issue was

```
▶ python -m trace -t atomicapp/cli/main.py -v --dry-run stop --provider=docker tests/units/cli/test_examples/helloapache | grep get_provider
2015-10-19 11:07:18,733 - atomicapp.plugin - DEBUG - Loading providers from /usr/local/lib/python2.7/dist-packages/atomicapp-0.1.11-py2.7.egg/atomicapp/providers
lib.py(57):     def get_provider(self, provider_key=None, dry=False):
base.py(111):         provider_key, provider = self.get_provider(provider_key, dryrun)
 --- modulename: lib, funcname: get_provider
2015-10-19 11:07:18,933 - root - DEBUG - GET-PROVIDER: True
2015-10-19 11:07:18,934 - atomicapp.plugin - DEBUG - Found provider <class 'docker.DockerProvider'>
2015-10-19 11:07:18,936 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
base.py(209):         provider_key, provider = self.get_provider(provider_key)
 --- modulename: lib, funcname: get_provider
2015-10-19 11:07:18,937 - root - DEBUG - GET-PROVIDER: False
2015-10-19 11:07:18,938 - atomicapp.plugin - DEBUG - Found provider <class 'docker.DockerProvider'>
2015-10-19 11:07:18,939 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
2015-10-19 11:07:18,940 - docker - DEBUG - DRY-RUN SETTING: False
2015-10-19 11:07:18,940 - docker - DEBUG - Given config: {u'image': u'centos/httpd', u'hostport': 80, 'namespace': 'default', 'provider': 'kubernetes'}
2015-10-19 11:07:18,941 - docker - DEBUG - Namespace: default
2015-10-19 11:07:18,958 - docker - INFO - Undeploying to provider: Docker
```

incoming PR!

**Comment by charliedrage** [comment]
fixed by #340
